src/rsa_code.rs

```diff
@@ -0,0 +1,113 @@
+// Implements http://rosettacode.org/wiki/RSA_code
+extern crate num;
+
+use num::bigint::BigUint;
+use num::traits::{Zero, One};
+use num::integer::Integer;
+#[cfg(not(test))]
+use std::str::FromStr;
+
+fn mod_exp(b: &BigUint, e: &BigUint, n: &BigUint) -> Result<BigUint, &'static str> {
+    if n.is_zero() {
+        return Err("modulus is zero");
+    }
+    if b >= n {
+        // base is too large and should be split into blocks
+        return Err("base is >= modulus");
+    }
+    if b.gcd(n) != BigUint::one() {
+        return Err("base and modulus are not relatively prime");
+    }
+
+    let mut bb = b.clone();
+    let mut ee = e.clone();
+    let mut result = BigUint::one();
+    while !ee.is_zero() {
+        if ee.is_odd() {
+            result = (result * &bb) % n;
+        }
+        ee = ee >> 1;
+        bb = (&bb * &bb) % n;
+    }
+    Ok(result)
+}
+
+#[cfg(not(test))]
+fn main() {
+    let msg = "Rosetta Code";
+
+    let n = BigUint::from_str("9516311845790656153499716760847001433441357").unwrap();
+    let e = BigUint::from_str("65537").unwrap();
+    let d = BigUint::from_str("5617843187844953170308463622230283376298685").unwrap();
```

Comment:
You could also do this as in `let e: BigUint = "65537".parse().unwrap()`, but of course it is just a question of style.